src/Microsoft.DotNet.Build.Tasks/GetPackageVersion.cs

```diff
@@ -34,7 +35,13 @@ public override bool Execute()
                 select el).FirstOrDefault();
 
             Debug.Assert(embeddedVerNode != null, "embeddedVerNode shouldn't be null");
-            PackageVersion = string.Format("{0}-{1}", embeddedVerNode.Value, BuildNumber);
+
+            // if this is a prerelease version then append the build number to the end
+            Regex preReleaseMatch = new Regex("-[0-9A-Za-z]+$");
```

Comment:
Per the SemVer spec any prerelease identifier is allowed to be alphanumeric but you are correct that today our prerelease identifiers don't contain any numeric characters.  As for moving the prerelease identifier out of the NuSpec file, I think this makes sense as long as every package built from a repo is in the same bucket (i.e. they are either all prerelease or not).